bench1/q4 vector CSUP query causes "panic: runtime error: index out of range [0] with length 0"

Repro is with super commit 1f0c0ce.

Per https://github.com/brimdata/super/pull/5507#issuecomment-2501949457, now that both #5507 and #5508 have merged, I'm revisiting the mgbench queries with CSUP to make sure they all still run ok and see if their performance has improved. When doing this, I've found that the [bench1/q4](https://github.com/brimdata/mgbench-private/blob/main/bench1/q4/q4.spq) query now panics where it didn't previously.

Test data is `s3://brim-sampledata/mgbench/bench1.csup`. The older CSUP that used to run ok via `super dev vector query` pre-#5507 has been renamed to `s3://brim-sampledata/mgbench/bench1-pre-super-5507.csup`.

```
$ super -version
Version: v1.18.0-180-g1f0c0cea

$ SUPER_VAM=1 super -c '
from "bench1.csup"
| where
    machine_group=="Servers"
    and cpu_wio > float32(0.99)
    and log_time >= 2016-12-01T00:00:00Z
    and log_time < 2017-01-01T00:00:00Z
| summarize
    spikes := count()
    by machine_name
| sort -r spikes
| head 10'

panic: runtime error: index out of range [0] with length 0

goroutine 2486 [running]:
github.com/brimdata/super/runtime/vam/op/summarize.(*countByString).countDict(0xc000b94200, 0xdb65d3c?, {0x0, 0x0, 0x1?}, 0x0)
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:204 +0x1a5
github.com/brimdata/super/runtime/vam/op/summarize.(*countByString).update(0xf4f9020?, {0xc000b941c0?, 0x0?, 0xc000a03c08?}, {0xc000b941d0?, 0x574792f8?, 0xc000b95f80?})
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:151 +0x134
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).consume(0xc0006c6d10, {0xc000b941c0, 0x1, 0x2}, {0xc000b941d0, 0x1, 0x1})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:102 +0x4cc
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull.func1({0xc000b941c0, 0x2, 0xc000a03d80?})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:83 +0x78
github.com/brimdata/super/vector.Apply(0x10?, 0xf78fa78?, {0xc000b941c0?, 0x2?, 0xf58c620?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc0006c6d10, 0x80?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:82 +0x13b
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0xc0011f0540)
	/Users/phil/work/super/runtime/vam/op/combine.go:110 +0x3f
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 2481
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

--- a/vector/view.go
+++ b/vector/view.go
@@ -19,6 +19,7 @@ func NewView(val Any, index []uint32) Any {
 		return NewConst(val.val, uint32(len(index)), NewBoolView(val.Nulls, index))
 	case *Dict:
 		index2 := make([]byte, len(index))
+		counts := make([]uint32, val.Any.Len())
 		var nulls *Bool
 		if val.Nulls != nil {
 			nulls = NewBoolEmpty(uint32(len(index)), nil)
@@ -26,14 +27,18 @@ func NewView(val Any, index []uint32) Any {
 				if val.Nulls.Value(idx) {
 					nulls.Set(uint32(k))
 				}
-				index2[k] = val.Index[idx]
+				v := val.Index[idx]
+				index2[k] = v
+				counts[v]++
 			}
 		} else {
 			for k, idx := range index {
-				index2[k] = val.Index[idx]
+				v := val.Index[idx]
+				index2[k] = v
+				counts[v]++
 			}
 		}
-		return NewDict(val.Any, index2, nil, nulls)
+		return NewDict(val.Any, index2, counts, nulls)
 	case *Error:
 		return NewError(val.Typ, NewView(val.Vals, index), NewBoolView(val.Nulls, index))
 	case *Union:

